### PR TITLE
implement multiclass dice loss

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

**Comment by Kaixhin** [comment]
Overall code looks fine, but would be good to have the code generalise to more than 3D. Moving forward I've seen more Torch code written to be generic for n-dimensional tensors where n > 2 (i.e., BxCxH onwards).

Looked around a bit and came across a more general version called the Tversky loss (see implementation and discussion of 3D Dice: https://github.com/keras-team/keras/issues/9395#issuecomment-379228094). Might be worth doing this and having Dice loss as a special case, so several things are covered in one bit of code?

**Comment by edgarriba** [comment]
@Kaixhin good point ! I think I'll directly import this https://github.com/kevinzakka/pytorch-goodies/blob/master/losses.py

**Comment by Kaixhin** [comment]
@edgarriba yep looks like the right kind of idea - perhaps all just in `losses\segmentation.py`, and then there'll be a nice selection to use from the start.
